### PR TITLE
feat(dapps) add max fees support for wallet connect requests

### PR DESCRIPTION
**Pull request by stefandunca**
### Updates: #15192
    
Compute max fees for transaction related requests and display the results to user.
    
Also:

- Add helper to convert from hex (backend) to decimal (frontend) values.
- Add helper to convert from float gwei to hex wei
- Update tests to accommodate for the new dependencies.
    
Sourcing of account balances is not included therefore the transaction is allowed to go through even if the account balance is insufficient. An error will be generated by the backend in this case.

</img>

### CODE REVIEW

File: ui/app/AppLayouts/Wallet/services/dapps/DAppsRequestHandler.qml [hunk 1]
```diff
@@ -324,28 +342,110 @@ QObject {
...
+        // }
+        function getEstimatedMaxFees(data, method, chainId) {
             var tx = {}
+            if (d.isTransactionMethod(method)) {
+                tx = d.getTxObject(method, data)
+            }
+
+            let Math = SQUtils.AmountsArithmetic
+            var gasLimit = Math.fromString("21000")
+            var gasPrice, maxFeePerGas, maxPriorityFeePerGas
```
Comment (Seitseman) [inline ui/app/AppLayouts/Wallet/services/dapps/DAppsRequestHandler.qml:374]: why not `let`?

**Comment by clauxx** [comment]
I think the fees should be formatted to 3-4 decimals, otherwise it looks weird. Also, is there a task to show the (relevant) transaction values in a human readable form (there must be an acceptance criteria for it) e.g. value, gas, fees, nonce, etc.?

**Comment by stefandunca** [comment]
> I think the fees should be formatted to 3-4 decimals, otherwise it looks weird.

Reported here https://github.com/status-im/status-desktop/issues/15612

> Also, is there a task to show the (relevant) transaction values in a human readable form (there must be an acceptance criteria for it) e.g. value, gas, fees, nonce, etc.?

Current task includes all this criteria also https://github.com/status-im/status-desktop/issues/15192

File: ui/imports/shared/stores/DAppsStore.qml [hunk 2]
```diff
@@ -87,6 +93,14 @@ QObject {
         return controller.getDapps()
     }
 
+    function hexToDec(hex) {
+        return controller.hexToDecBigString(hex)
+    }
+
```
Comment (alexjba) [inline ui/imports/shared/stores/DAppsStore.qml:99]: Would be nice to describe the expected json format

File: ui/app/AppLayouts/Wallet/services/dapps/DAppsRequestHandler.qml [hunk 3]
```diff
@@ -324,28 +342,110 @@ QObject {
...
+            let maxFeesEth = Math.div(feesInfo.maxFees, Math.fromString("1000000000"))
+
+            // TODO #15192: extract account.balance
+            //let accountFundsEth = account.balance
+            //let haveEnoughFees = Math.cmp(accountFundsEth, maxFeesEth) >= 0
+            let haveEnoughFees = true
+
+            let maxFeesEthStr = maxFeesEth.toString()
+            let fiatMaxFees = root.currenciesStore.getFiatValue(maxFeesEthStr, Constants.ethToken)
+            let symbol = root.currenciesStore.currentCurrencySymbol
```
Comment (alexjba) [inline ui/app/AppLayouts/Wallet/services/dapps/DAppsRequestHandler.qml:427]: I think we should use the currency short name. Sorry, probably the Sign modal API is misleading here.

**Review by alexjba — APPROVED** [review]
LGTM with minor finding on the currency symbol! Tested and works nicely! 👍

**Pull request merged**